### DISCUSSION TITLE
Issue regarding the header anchor

### DISCUSSION THREAD

**Discussion by clecho** [vuepress/core#471] [Q&A] [answered]
I have the following issue using vuepress (version v2.0.0-beta.26). On refresh, the page does not scroll to the anchor selected. I stay at the top of the page, and the anchor selected in the URL disappears. 

To sum up:
- I scroll down to reach a header anchor
- the anchor appears in the URL
- refresh the page.
- The page does not scroll to the anchor, instead I stay on the top of the page and the anchor disappears.

My question would be: do you have any idea why I get this behavior? Or do you know how i can investigate this issue?

I try the following:
- Try to make it work on a very simple page
- Remove any suspicious element in my configuration (config.ts)

Another anomaly that I encountered (that may be linked with the anchor issue), on refresh, the sidebar disappears (it displays correctly when I navigate from the root of my website)

**Comment by clecho** [comment]
I ended up adding a bit of javascript code to fix my anchor issue:

`config.ts`
````typescript
{
    head: [
        ['script', {}, `
          (function() { 
            var hash = window.location.hash;
            setTimeout(()=>{
              const element = document.querySelector(hash)
              const topPos = element.getBoundingClientRect().top + window.pageYOffset
              window.scrollTo({
                top: topPos,
                behavior: 'smooth'
              })
            }, 100)
          })(); 
    `]
    ]
}
````

Credits:
- https://github.com/vuejs/vuepress/issues/790#issuecomment-418107492
- https://stackoverflow.com/a/52138511

**Comment by Mister-Hope** [comment, marked as answer]
A related issue is created, see #693, and I make a PR #722. Also, your answer is a good workaround.👍
